### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by Clockknight**
Partner: @slanante

Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Extra Credit
Issue 3: Feature request implemented ✔
Issue 3: Feature request unit tested ✔

### CODE REVIEW

**Comment by slanante** [comment]
Will do a review

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public static void Main(string[] args)
 
         public static string GetPercentCorrect(int numberCorrectAnswers, int numberOfQuestions)
         {
-            return (numberCorrectAnswers / numberOfQuestions * 100) + "%";
+            return  ((( (float) numberCorrectAnswers / (float) numberOfQuestions) * 100).ToString()) + "%";
```
Comment (slanante) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:27]: Looks great! All tests pass and program runs as expected. Great idea for the EC portion. The only revision I can think of is cleaning the GetPercentCorrect method with this:
```suggestion
          return ((float)numberCorrectAnswers / numberOfQuestions * 100) + "%";
```
Typecasting the numerator as a float allows for the correct result. ToString() is unnecessary as the concatenation of "%" will automatically convert the the given float to a string.

Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Extra Credit
Issue 3: Feature request implemented ✔
Issue 3: Feature request unit tested ✔

**Review by slanante — COMMENTED** [review]
Thanks again for pairing with me, I apologize for not being able to contribute more due to my job outside of school. See you later :)

**Comment by Wvvern** [comment]
Will review

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 2]
```diff
@@ -6,6 +6,28 @@ namespace PrincessBrideTrivia.Tests
     [TestClass]
     public class ProgramTests
     {
+        
+
+        [TestMethod]
+        [DataRow("1")]
+        [DataRow("2")]
+        [DataRow("3")]
+        public void CheatCodeAlwaysReturnsTrue(string correctIndex)
```
Comment (Wvvern) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:15]:
```suggestion
        public void DisplayResult_GivenCheatCode_ReturnsTrue(string correctIndex)
```
The readability here could be improved by following the recommended naming convention for tests specified here:
https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices

```suggestion
        public void DisplayResult_ReturnsTrueIfGivenCheatCode(string correctIndex)
```
Alternatively, you could name it according to the convention already set within the code.
Such as the existing method: "DisplayResult_ReturnsTrueIfCorrect()"
Although, I do prefer the convention specified by microsoft.

Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

Looks great! You should consider [slanante's](https://github.com/slanante) suggestion for GetPercentCorrect(). You could also round to the nearest two decimal points there. Otherwise, the precision of the decimal number is simply ridiculous and only serves to clutter the screen.

**Comment by BenagainB** [comment]
Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Extra Credit
Issue 3: Feature request implemented ✔
Issue 3: Feature request unit tested ✔

Everything looks to be running well. The cheat code is a good extra feature, since you can use it while debugging, but you might want to disable the cheat code by default as an extra layer of protection. Also, you might think about only accepting answers in the range [1,3] and looping until the user gives a valid number or an exit code to leave the game early. Good job!

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 3]
```diff
@@ -6,6 +6,28 @@ namespace PrincessBrideTrivia.Tests
     [TestClass]
     public class ProgramTests
     {
+        
+
```
Comment (JLee35) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:10]: Eliminating unnecessary whitespace like this is low hanging fruit for making your code cleaner.

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs (hunk 1 above)
Comment (JLee35) [reply PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:27]: Agreed that this could be cleaned up, but I do like how the output is formatted to the whole percent.

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 4]
```diff
@@ -42,7 +42,8 @@ public static string GetGuessFromUser()
 
         public static bool DisplayResult(string userGuess, Question question)
         {
-            if (userGuess == question.CorrectAnswerIndex)
+            String cheatcode = "idontknowtheanswer";
+            if (userGuess == question.CorrectAnswerIndex || userGuess.Equals(cheatcode))
```
Comment (JLee35) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:46]: Nice touch!

**Comment by JLee35** [comment]
Remember to do code reviews on other PRs that are not your own and acknowledge peer reviews.
